src/orderbook.js

```diff
@@ -1204,20 +1343,24 @@ OrderBook.prototype.computeAutobridgedOffers = function() {
  */
 
 OrderBook.prototype.mergeDirectAndAutobridgedBooks = function() {
-  const self = this;
+
+  if (this._destroyed) {
+    return;
+  }
 
   if (_.isEmpty(this._offers) && _.isEmpty(this._offersAutobridged)) {
+    if (this._synchronized && this._gotOffersFromLegOne &&
+      this._gotOffersFromLegTwo) {
+      // emit empty model to indicate to listeners that we've got offers,
+      // just there was no one
+      this.emit('model', []);
+    }
     return;
   }
 
   this._mergedOffers = this._offers
     .concat(this._offersAutobridged)
-    .sort(function(a, b) {
-      const aQuality = OrderBookUtils.getOfferQuality(a, self._currencyGets);
-      const bQuality = OrderBookUtils.getOfferQuality(b, self._currencyGets);
-
-      return aQuality.compareTo(bQuality);
-    });
+    .sort(_sortOffers.bind(this));
```

Comment:
Another option would be to create the amounts once separate from the `mergedOffers` array, sort and then remap the offers array

```
  const offers = self._offers.concat(self._offersAutobridged);
  const qualities = offers.map(function(o, i) {
    return {
      index: i,
      value: OrderBookUtils.getOfferQuality(o, self._currencyGets)
    }
  });

  qualities.sort(function(a, b) {
    return a.value.compareTo(b.value);
  });

  self._mergedOffers = qualities.map(function(q) {
    return offers[q.index];
  });
```